modules/apps/segments/segments-web/src/main/java/com/liferay/segments/web/internal/display/context/SegmentsExperienceSelectorDisplayContext.java

```diff
@@ -79,28 +81,52 @@ public JSONArray getSegmentsExperiencesJSONArray() throws PortalException {
 		return segmentsExperiencesJSONArray;
 	}
 
-	public String getSelectedSegmentsExperienceName() {
-		long segmentsExperienceId = ParamUtil.getLong(
-			_httpServletRequest, "segmentsExperienceId", -1);
+	public String getSegmentsExperienceStatusLabel(boolean active) {
+		String statusLabelKey = "inactive";
 
-		if (segmentsExperienceId == -1) {
-			segmentsExperienceId =
-				_segmentsExperienceManager.getSegmentsExperienceId(
-					_httpServletRequest);
+		if (active) {
+			statusLabelKey = "active";
 		}
 
+		return LanguageUtil.get(_httpServletRequest, statusLabelKey);
+	}
+
+	public String getSelectedSegmentsExperienceName() {
 		SegmentsExperience segmentsExperience =
-			SegmentsExperienceLocalServiceUtil.fetchSegmentsExperience(
-				segmentsExperienceId);
+			_fetchSegmentsExperienceFromRequest();
 
 		SegmentsExperience parentSegmentsExperience =
 			_getParentSegmentExperience(segmentsExperience);
 
-		if (parentSegmentsExperience != null) {
+		if ((segmentsExperience != null) &&
+			(parentSegmentsExperience != null)) {
+
 			segmentsExperience = parentSegmentsExperience;
 		}
 
-		return segmentsExperience.getName(_themeDisplay.getLocale());
+		if (segmentsExperience != null) {
+			return segmentsExperience.getName(_themeDisplay.getLocale());
+		}
+
+		return StringPool.BLANK;
+	}
+
+	public boolean isSelectedSegmentsExperienceActive(
+		JSONArray segmentsExperiencesJSONArray) {
+
+		SegmentsExperience segmentsExperience =
+			_fetchSegmentsExperienceFromRequest();
+
+		if ((segmentsExperience != null) &&
+			(segmentsExperience.getSegmentsExperienceId() ==
+				_getFirstAssociatedSegmentsExperienceId(
+					segmentsExperience.getSegmentsEntryId(),
+					segmentsExperiencesJSONArray))) {
+
+			return true;
+		}
+
+		return false;
 	}
 
 	private void _calculateActiveSegmentsExperiencesJSONArray(
```

Comment:
instead using _calculateActiveSegmentsExperiencesJSONArray, could we add the "active" label in _getSegmentsExperienceJSONObject directly?